datus/cli/web/ui_components.py

```
@@ -252,27 +284,271 @@ def show_available_subagents(self, agent_config) -> None:
+        else:
+            with tab2:
+                st.error("SQL execution failed")
+                st.markdown(data.error)
+            with tab3:
+                st.error("SQL execution failed")
+                st.markdown(data.error)
+        return sql_id
+
+    def render_dynamic_chart_in_tab(self, df: pd.DataFrame, state_id: str, chart_type: str):
+        """
+        在 Tab 内部渲染一个可配置的动态图表，配置项位于 st.popover 中。
```

Ruff (0.14.8)

Docstring contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF002)